field_RANKSUFF_module.fypp

```diff
@@ -52,6 +52,8 @@ CONTAINS
   PROCEDURE :: SYNC_DEVICE_RDONLY => ${ftn}$_SYNC_DEVICE_RDONLY
   PROCEDURE :: COPY_OBJECT => ${ftn}$_COPY_OBJECT
   PROCEDURE :: WIPE_OBJECT => ${ftn}$_WIPE_OBJECT
+  PROCEDURE(GET_DIMS), DEFERRED :: GET_DIMS
```

Comment:
This looks cool, but pretty advanced OO usage. Is this device-safe? Has this been checked? Not a showstopper for me, as we don't use FIELD objects on device yet; I was just curious how far NVHPC supports this.

Reply:
It's not possible to use the deferred method on the device, but I think it's not possible to use any type-bound method on the device anyway. Those methods are meant to be used on the host, in preparation of the compute kernels executing on the device. That being said, it is just a limitation of the openacc implementation, it would make sense to be able to call some methods on the device, like get_dims for instance.